include/xgboost/c_api.h

```diff
@@ -126,12 +126,29 @@ XGB_DLL int XGBGetGlobalConfig(char const **out_config);
 
 /*!
  * \brief load a data matrix
+ * \deprecated since 1.7.3
```

Comment:
I think the next big one would be 2.0 unless something major comes up.

Reply:
Done.